src/update-feeds-loops/get-updatable-feeds.ts

```diff
@@ -19,122 +19,152 @@ export interface UpdatableBeacon {
-        //    beacon set value or timestamp to change. See:
-        //    https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/DataFeedServer.sol#L54
-        const isBeaconSet = beaconValues.length > 1;
-        // Note, that the beacon set value/timestamp is computed as median, so single beacon update may not result in a beacon set update.
-        if (isBeaconSet && newDataFeedValue === dataFeedValue && newDataFeedTimestamp === dataFeedTimestamp) {
-          return false;
-        }
-        if (!isBeaconSet && newDataFeedTimestamp === dataFeedTimestamp) return false;
-
-        return logger.runWithContext(
-          {
-            dapiName: decodedDapiName,
-            dataFeedId,
-          },
-          () =>
-            checkUpdateCondition(
-              dataFeedValue,
-              dataFeedTimestamp,
-              newDataFeedValue,
-              newDataFeedTimestamp,
-              adjustedHeartbeatInterval,
-              adjustedDeviationThresholdCoefficient,
-              deviationReference
-            )
-        );
-      })
-      // Compute the updateable beacons.
-      .map(({ dataFeedInfo, aggregatedBeaconsWithData }) => ({
+        },
+        () =>
+          checkUpdateCondition(
+            dataFeedValue,
+            dataFeedTimestamp,
+            newDataFeedValue,
+            newDataFeedTimestamp,
+            adjustedHeartbeatInterval,
+            adjustedDeviationThresholdCoefficient,
+            deviationReference
+          )
+      );
+
+    if (dataFeedNeedsUpdate) {
+      acc.push({
         dataFeedInfo,
         updatableBeacons: aggregatedBeaconsWithData
           .filter(({ isUpdatable }) => isUpdatable)
           .map(({ beaconId, signedData }) => ({
             beaconId,
             signedData: signedData!,
           })),
-      }))
-  );
+        shouldUpdateBeaconSet: isBeaconSet,
+      });
+    } else if (isBeaconSet && asyncBeaconUpdatedeviationThresholdFactor) {
+      // 2.5. There is a special case when data feed is a beacon set that do not need to be updated but some of its beacon constituents do.
+      //      In this particular case, airseeker can update only these beacons and skip the beacon set update. This is enabled by setting a
+      //      value in asyncBeaconUpdatedeviationThresholdFactor on the config.
+      const updatableBeacons = aggregatedBeaconsWithData
+        .filter(
+          ({ isUpdatable, offChainValue, onChainValue }) =>
+            isUpdatable &&
+            offChainValue &&
+            checkUpdateCondition(
+              onChainValue.value,
+              onChainValue.timestamp,
+              offChainValue.value,
+              offChainValue.timestamp,
+              adjustedHeartbeatInterval,
+              adjustedDeviationThresholdCoefficient * BigInt(asyncBeaconUpdatedeviationThresholdFactor),
```

Comment:
I think this is too complex. I think it's better to make it a percentage and apply the adjustment factor like I mentioned in the other comment.

Reply:
I think I need more details on what's too complex here. Are you saying that calling checkUpdateCondition() inside an Array.filter() then Array.map() is to complex or that we use a factor of the adjustedDeviationThresholdCoefficient instead of a hardcoded percentage is too complex? If it is the later, then I guess the idea is that individual beacon update requirement is more relaxed than the beacon set threshold. For example, if beaconSet should update at 1% deviation then individual beacon update should only update when 1% is not reached but some beacon has reached 5% deviation. If we make the config option a percentage then I guess we should validate at least that it is higher than adjustedDeviationThresholdCoefficient with zod